### PR TITLE
Feature/fill in the blank puzzle

### PR DESCRIPTION
**Pull request by Alperdec**
Problem
=======
[Jira](https://cruzhacks-dev.atlassian.net/browse/C2D-180)

Solution
========
What I/we did to solve this problem
added a modal component and used it to make a popup puzzle when you click CruzHacks 2022

Change Summary:
---------------
* Tidy, well formulated commit message
* Another great commit message
* Something else I/we did

Dev-Ops
=======
If you added an ENV variable, please check off that you've updated the following  
- [ ] Key & Sample value to `.env` & `sample.env` 
- [ ] GCP Secret Manager (Both development and production)
- [ ] Github Secrets (Added a development and production variable)
- [ ] Github Workflows `.github/workflows/dev.yml` & `.github/workflows/prod.yml`
- [ ] webpack-config.js

Images/Important Notes (optional):
-----------------------
* Insert any notes/issues, dependencies added or removed, or images

### CODE REVIEW

**Review by krishnakpandian — COMMENTED** [review]
Code-wise this looks good, but I'm going to hold off on reviewing the UI and functionality for this and let Marketing check it out

**Comment by Alperdec** [comment]
@krishnakpandian Sounds good to me

File: src/views/Home/components/Landing/index.tsx
Comment (krishnakpandian) [inline src/views/Home/components/Landing/index.tsx]: Could we update this Code so it passes in that div as children and still retains this property. Main reason I'm asking this is so once CruzHacks is over and we close some services, it will be easier to undo our changes.

File: src/components/PuzzleModal/index.tsx
Comment (krishnakpandian) [inline src/components/PuzzleModal/index.tsx]: Let's use the React typing for these events

File: src/components/PuzzleModal/index.tsx
Comment (krishnakpandian) [inline src/components/PuzzleModal/index.tsx]: Let's update this message to “That's correct! Click here to earn your CruzCoins points!”

File: src/components/PuzzleModal/index.tsx
Comment (krishnakpandian) [inline src/components/PuzzleModal/index.tsx]: Update this message to “Sorry, that's incorrect. Guess again”

File: src/components/PuzzleModal/index.scss
Comment (krishnakpandian) [inline src/components/PuzzleModal/index.scss]: Let's increase the size of the modal just a little bit (maybe 5-10% bump in height and width.

**Review by krishnakpandian — CHANGES_REQUESTED** [review]
Here is some additional feedback from design but overall looks good! I did add a couple thoughts of my own in here as well

**Review by krishnakpandian — COMMENTED** [review]
It looks better! Let's see if we can make this more modular so that we can reuse it more and also add or remove data. I created a sample of this [here](https://github.com/CruzHacks/cruzhacks-2022-website/pull/154) which you can check out. I essentially made some tweaks to have it pass a React Node since C2D-181 will be a bit more complex in using the Puzzle function. Look through it, and update it as you like but this function should be quick and easy to add and remove!

File: src/components/PuzzleModal/index.scss
Comment (krishnakpandian) [inline src/components/PuzzleModal/index.scss:8]: pop-up-puzzle-text

Can we decrease the margin/padding here?

File: src/components/PuzzleModal/index.tsx
Comment (krishnakpandian) [inline src/components/PuzzleModal/index.tsx]: I would double check this for ' vs "

**Review by krishnakpandian — CHANGES_REQUESTED** [review]

File: src/components/PuzzleModal/index.tsx
Comment (Alperdec) [reply src/components/PuzzleModal/index.tsx]: sorry! good catch

File: src/components/PuzzleModal/index.scss
Comment (Alperdec) [reply src/components/PuzzleModal/index.scss:8]: I could decrease the padding on the inner elements, but I use the margin properties there to center it in the screen

File: src/components/PuzzleModal/index.tsx
Comment (krishnakpandian) [inline src/components/PuzzleModal/index.tsx]: So we have a space before and after in the `Click Here` let's remove those and add a space after "That's correct!" and "to earn your CruzCoins points!". Main reason is that the styles for the href are still being applied to those spaces.

File: src/components/PuzzleModal/index.tsx
Comment (krishnakpandian) [inline src/components/PuzzleModal/index.tsx]: pop-up-puzzle-sizing

The sizing still seems to be a little awkard for the success message.

Comment (krishnakpandian) [inline src/components/PuzzleModal/index.tsx]: Let's rename this class to be consistent

**Review by krishnakpandian — COMMENTED** [review]
I think it's almost there just needs a refinement on the sizing + spacing of the component.

**Comment by krishnakpandian** [comment]
Besides the slight sizing refinement, both Marketing and I are happy with how it is functioning. @rvaidun I would start your story of this branch and PR into this. Obviously adjust the puzzle component as needed to make it work for both the two of you but it's looking good!

**Comment by Alperdec** [comment]
@krishnakpandian I tried to make it look a little better. Let me know how this one looks

**Comment by krishnakpandian** [comment]
> @krishnakpandian I tried to make it look a little better. Let me know how this one looks

I think this seems good! I'll approve this once we get the other marketing challenge merged into this branch and that's all tested!

**Comment by Alperdec** [comment]
> > @krishnakpandian I tried to make it look a little better. Let me know how this one looks
> 
> I think this seems good! I'll approve this once we get the other marketing challenge merged into this branch and that's all tested!

Awesome, thanks for your patience :) I apologize for the kinks

**Review by w-gao — COMMENTED** [review]
Looks great, thanks @Alperdec! Will re-review once the other challenges are merged in.

File: src/components/PuzzleModal/index.tsx
Comment (w-gao) [inline src/components/PuzzleModal/index.tsx]: Also, I wanted to ask if this should've been blue?

Comment (Alperdec) [reply src/components/PuzzleModal/index.tsx]: Good point, I changed it to white but past that I'm not too sure what we want stying wise. I can definitely use an icon instead too

**Review by krishnakpandian — APPROVED** [review]
I'm going to approve this, I'm not sure if we will receive the other puzzle component today. If not, I feel good about what we have now and we can play the rest by ear over the next week in terms of development.

**Pull request merged**
